cmd/repo-updater/repos/store.go

```diff
@@ -429,6 +682,115 @@ func listReposQuery(args StoreListReposArgs) paginatedQuery {
 	}
 }
 
+func (s DBStore) UpsertSources(ctx context.Context, inserts, updates, deletes map[api.RepoID][]SourceInfo) error {
+	type source struct {
+		ExternalServiceID int64  `json:"external_service_id"`
+		RepoID            int64  `json:"repo_id"`
+		CloneURL          string `json:"clone_url"`
+	}
+
+	marshalSourceList := func(sources map[api.RepoID][]SourceInfo) ([]byte, error) {
+		srcs := make([]source, 0, len(sources))
+		for rid, infoList := range sources {
+			for _, info := range infoList {
+				srcs = append(srcs, source{
+					ExternalServiceID: info.ExternalServiceID(),
+					RepoID:            int64(rid),
+					CloneURL:          info.CloneURL,
+				})
+			}
+		}
+		return json.Marshal(srcs)
+	}
+
+	insertedSources, err := marshalSourceList(inserts)
+	if err != nil {
+		return err
+	}
+
+	updatedSources, err := marshalSourceList(updates)
+	if err != nil {
+		return err
+	}
+
+	deletedSources, err := marshalSourceList(deletes)
+	if err != nil {
+		return err
+	}
+
+	q := sqlf.Sprintf(upsertSourcesQueryFmtstr,
+		sqlf.Sprintf("%s", string(deletedSources)),
+		sqlf.Sprintf("%s", string(updatedSources)),
+		sqlf.Sprintf("%s", string(insertedSources)))
```

Comment:
Are `sqlf.Sprintf`s required? Don't seem so.
```suggestion
	q := sqlf.Sprintf(upsertSourcesQueryFmtstr,
		string(deletedSources),
		string(updatedSources),
		string(insertedSources),
	)
```

Reply:
Indeed, that seems so obvious now 😆